aicsimageio/readers/bioformats_reader.py

```diff
@@ -0,0 +1,442 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+from __future__ import annotations
+
+from functools import lru_cache
+from threading import Lock
+from typing import TYPE_CHECKING, Any, Dict, NamedTuple, Optional, Tuple, Union
+
+import dask.array as da
+import numpy as np
+import xarray as xr
+from fsspec.implementations.local import LocalFileSystem
+from ome_types import OME
+
+from .. import constants, dimensions, exceptions
+from ..metadata import utils as metadata_utils
+from ..utils import io_utils
+from ..utils.cached_property import cached_property
+from ..utils.dask_proxy import DaskArrayProxy
+from .reader import Reader
+
+if TYPE_CHECKING:
+    from fsspec.spec import AbstractFileSystem
+
+    from .. import types
+
+
+try:
+    import jpype
+    from bioformats_jar import get_loci
+except ImportError:
+    raise ImportError(
+        "bioformats_jar is required for this reader. "
+        "Install with `pip install aicsimageio[bioformats]`"
+    )
+
+
+class BioformatsReader(Reader):
+    """Read files using bioformats.
+
+    This reader requires bioformats_jar to be installed in the environment, and requires
+    the java executable to be available on the path, or via the JAVA_HOME environment
+    variable.
+
+    Parameters
+    ----------
+    image : Path or str
+        path to file
+
+    Raises
+    ------
+    exceptions.UnsupportedFileFormatError
+        If the file is not supported by bioformats.
+    """
+
+    @staticmethod
+    def _is_supported_image(fs: AbstractFileSystem, path: str, **kwargs: Any) -> bool:
+        try:
+            if not isinstance(fs, LocalFileSystem):
+                return False
+            f = LociFile(path, meta=False, memoize=False)
+            f.close()
+            return True
+
+        except Exception:
+            return False
+
+    def __init__(self, image: types.PathLike):
+        self._fs, self._path = io_utils.pathlike_to_fs(image, enforce_exists=True)
+        # Catch non-local file system
+        if not isinstance(self._fs, LocalFileSystem):
+            raise ValueError(
+                f"Cannot read Bioformats from non-local file system. "
+                f"Received URI: {self._path}, which points to {type(self._fs)}."
+            )
+
+        try:
+            with LociFile(self._path) as rdr:
+                self._scenes: Tuple[str, ...] = tuple(
+                    metadata_utils.generate_ome_image_id(i)
```

Comment:
I think the scene names could actually come from the metadata, no?  Now I'm second guessing, do we want to enforce that they are these ome_image_ids or the names coming from the file?

Reply:
yeah, wasn't quite sure what to do here either!  happy to take your advice on that one.  @JacksonMaxfield?

Reply:
I had to think for a sec, but as a user I might already know the scene names I put in the microscopy file and so might want to reference them as such.

Reply:
If there is an easy loci one liner then yes I would personally pull the scene names out from the raw file.

Reply:
> If there is an easy loci one liner then yes I would personally pull the scene names out from the raw file.

so just to confirm, do you guys mean the "name" field of each top level Image (for OME).  or the proprietary scene name?  If we take, for example, the `s_3_t_1_c_3_z_5.czi` test file (where the "CziReader" would yield scenes named "P1, P2, P3") ...  the Image "Name" attributes look like this:

```xml
<Image ID="Image:0" Name="s_3_t_1_c_3_z_5.czi #1">
<Image ID="Image:1" Name="s_3_t_1_c_3_z_5.czi #2">
<Image ID="Image:1" Name="s_3_t_1_c_3_z_5.czi #3">
```
(on leica, those Name fields actually _do_ match what you extract for scene names)

the zeiss P1/2/3 strings are buried in the original metadata structured annotations (which would be a pain to extract for each format):

```xml
        <XMLAnnotation ID="Annotation:668" Namespace="openmicroscopy.org/OriginalMetadata">
            <Value>
                <OriginalMetadata>
                    <Key>Information|Image|S|Scene|Name</Key>
                    <Value>[P2, P3, P1]</Value>
                </OriginalMetadata>
            </Value>
        </XMLAnnotation>
```


so should we A) just use `Image:0` etc... or B) use the `Name` of the OME Image attribute?

Reply:
Option B seems the most likely to be what the user would have named their scenes, I would hope?

Reply:
(Does bioformats always put something fake there or is it at least attempting to fish out the real names?)

Reply:
Yea This has bee a bad spot in the API.. I think some readers currently return what is the "ID" and some readers currently return what is the "name" here. Personally, I am fine with returning the ID (as you have done) as I think a majority of our readers currently return ID... We should make a decision on that later though.

If @toloudis agrees with my above sentiment I can file an issue for us to standardize our ID or Name selection later.

Reply:
My sense is that Name is a better guess for what a user would pass as input, and a more sensible scene name for our API to return.   Those ID things are more like an integer index and if bioformats is actually extracting name info from the PFF's metadata, then it would put it in the Name field and not the ID field.

Reply:
I think the only implication on this is that you have to get the ome metadata here, but that is no worse than what other readers would have to do.

Reply:
That's fair. I guess using name does make more sense. Even if the names are gross like the example you provided.

Reply:
will do

Reply:
updated.  they _are_ pretty gross 😂 ... (except for leica, which does get better)